staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/controller.go

```diff
@@ -89,6 +90,30 @@ func (c *Controller) Run(staticSpec *spec.Swagger, openAPIService *handler.OpenA
 		return
 	}
 
+	// create initial spec to avoid merging once per CRD on startup
+	crds, err := c.crdLister.List(labels.Everything())
+	if err != nil && !errors.IsNotFound(err) {
+		utilruntime.HandleError(fmt.Errorf("failed to initially list all CRDs: %v", err))
+		return
+	}
+	for _, crd := range crds {
+		if !apiextensions.IsCRDConditionTrue(crd, apiextensions.Established) {
+			continue
+		}
+		newSpecs, changed, err := buildVersionSpecs(crd, nil)
+		if err != nil {
+			klog.Warningf("failed to build OpenAPI spec of CRD %s: %v", crd.Name, err)
```

Comment:
use `utilruntime.HandleError` instead of klog to be consistent with the [handling](https://github.com/kubernetes/kubernetes/blob/df2094b3d728bd58c5f23b41add109e32fc7c301/staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/controller.go#L125) used by [sync](https://github.com/kubernetes/kubernetes/blob/df2094b3d728bd58c5f23b41add109e32fc7c301/staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/controller.go#L119)

nit: could be weird if `changed` and `err` were both set. Maybe add `continue` here as well?

Reply:
feel free to tag this PR if `klog` is preferable to `utilruntime.HandleError`

Reply:
fixed